src/Initializer/InitProcedure/Init.cpp

```diff
@@ -9,22 +9,24 @@
 #include <Numerical_aux/Statistics.h>
 #include "ResultWriter/ThreadsPinningWriter.h"
 #include <sstream>
+#include "Monitoring/Unit.hpp"
 
 namespace {
 
 static void reportDeviceMemoryStatus() {
 #ifdef ACL_DEVICE
   device::DeviceInstance& device = device::DeviceInstance::getInstance();
-  constexpr size_t GB = 1024 * 1024 * 1024;
   const auto rank = seissol::MPI::mpi.rank();
   if (device.api->getCurrentlyOccupiedMem() > device.api->getMaxAvailableMem()) {
     std::stringstream stream;
 
     stream << "Device(" << rank << ")  memory is overloaded."
-           << "\nTotally allocated device memory, GB: "
-           << device.api->getCurrentlyOccupiedMem() / GB << "\nAllocated unified memory, GB: "
-           << device.api->getCurrentlyOccupiedUnifiedMem() / GB
-           << "\nMemory capacity of device, GB: " << device.api->getMaxAvailableMem() / GB;
+           << "\nTotally allocated device memory: "
+           << UnitByte.formatPrefix(device.api->getCurrentlyOccupiedMem()).c_str()
+           << "\nAllocated unified memory: "
+           << UnitByte.formatPrefix(device.api->getCurrentlyOccupiedUnifiedMem()).c_str()
+           << "\nMemory capacity of device: "
+           << UnitByte.formatPrefix(device.api->getMaxAvailableMem()).c_str();
```

Comment:
That shouldn't matter (and it may be a bit worse, cf. e.g. here https://stackoverflow.com/questions/12675273/why-does-using-stdendl-with-ostringstream-affect-output-speed ). ... However, I agree for readability.

Reply:
Ok.